src/legacy/core_plugins/vis_type_xy/public/plugin.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {
+  CoreSetup,
+  CoreStart,
+  Plugin,
+  IUiSettingsClient,
+  PluginInitializerContext,
+  MountPoint,
+  UnmountCallback,
+} from 'kibana/public';
+
+import { Plugin as ExpressionsPublicPlugin } from '../../../../plugins/expressions/public';
+import { VisualizationsSetup, VisualizationsStart } from '../../visualizations/public';
+import { ChartsPluginSetup } from '../../../../plugins/charts/public';
+
+export interface VisTypeXyDependencies {
+  uiSettings: IUiSettingsClient;
+  charts: ChartsPluginSetup;
+  mountBanner: (mount: MountPoint<HTMLElement>) => UnmountCallback;
+}
+
+/** @internal */
+export interface VisTypeXyPluginSetupDependencies {
+  expressions: ReturnType<ExpressionsPublicPlugin['setup']>;
+  visualizations: VisualizationsSetup;
+  charts: ChartsPluginSetup;
+}
+
+/** @internal */
+export interface VisTypeXyPluginStartDependencies {
+  expressions: ReturnType<ExpressionsPublicPlugin['start']>;
+  visualizations: VisualizationsStart;
+}
+
+type VisTypeXyCoreSetup = CoreSetup<VisTypeXyPluginStartDependencies>;
+
+/** @internal */
+export class VisTypeXyPlugin implements Plugin<Promise<void>, void> {
+  constructor(public initializerContext: PluginInitializerContext) {}
+
+  public async setup(
+    core: VisTypeXyCoreSetup,
+    { expressions, visualizations, charts }: VisTypeXyPluginSetupDependencies
+  ) {
+    const [{ overlays }] = await core.getStartServices();
+    const mountBanner: VisTypeXyDependencies['mountBanner'] = mountBannerOnce(overlays);
```

Comment:
Looks like the banner isn't mounted anymore because it's not actually overwriting a visualization. That's fine for me in general, just wanted to make sure it's happening on purpose.

Reply:
Yup, I figured just leaving it in their until the first replacement visualization is added and I'll mount it in there.